jdtls.ext/com.microsoft.jdtls.ext.core/src/com/microsoft/jdtls/ext/core/PackageCommand.java

```diff
@@ -354,6 +354,25 @@
         return Collections.emptyList();
     }
 
+    private static IPackageFragmentRoot getPackageFragmentRootFromQuery(PackageParams query) {
+        IPackageFragmentRoot packageRoot = (IPackageFragmentRoot) JavaCore.create(query.getHandlerIdentifier());
+        if (packageRoot != null) {
+            return packageRoot;
+        }
+
+        // jar in Referenced Libraries must be constructed from path
+        IJavaProject javaProject = getJavaProject(query.getProjectUri());
+        if (javaProject != null) {
+            try {
+                packageRoot = javaProject.findPackageFragmentRoot(Path.fromPortableString(query.getRootPath()));
```

Comment:
What is the difference between `fromOSString()` and `fromPortableString()` here?

Reply:
`fromPortableString()` is used here because the path is generated by `toPortableString()`

See: https://github.com/microsoft/vscode-java-dependency/blob/master/jdtls.ext/com.microsoft.jdtls.ext.core/src/com/microsoft/jdtls/ext/core/model/PackageNode.java#L252